src/errors/errors.c

```diff
@@ -0,0 +1,68 @@
+#include <errors/errors.h>
+#include <stdio.h>
+#include <defines.h>
+#include <stdlib.h>
+#include <string.h>
+#include <errno.h>
+#include <output/write_to_std.h>
+#include <executor/run_commands.h>
+
+static t_bool	should_exit(t_exit_code code)
+{
+	if (code == MALLOC_ERROR || \
+		code == DUP_ERROR || \
+		code == FORK_ERROR || \
+		code == PIPE_ERROR)
+	{
+		set_exit_code(SYS_ERROR);
```

Comment:
Bash never exits with a minus code, i think we can set the exit code to 1 if there is a system call error.